### PR TITLE
Eslint - Add no-self-assign rule

### PR DESCRIPTION
**Pull request by San4es**
Info: https://eslint.org/docs/rules/no-self-assign

### CODE REVIEW

File: testing/tests/DevExpress.viz.charts/chart.part4.tests.js [hunk 1]
```diff
@@ -1248,8 +1248,8 @@ QUnit.test("Three axis without names", function(assert) {
         chart = this.createChart({
             valueAxis: [{}, {}, {}],
 
-            incidentOccurred: function(message) {
-                message = message;
```
Comment (nightskylark) [inline testing/tests/DevExpress.viz.charts/chart.part4.tests.js:1252]: It was awful.

**Review by nightskylark — DISMISSED** [review]

File: js/data/odata/utils.js [hunk 2]
```diff
@@ -363,7 +363,6 @@ var interpretVerboseJsonFormat = function(obj) {
         return { error: Error("Malformed or unsupported JSON response received") };
     }
 
-    data = data;
```
Comment (nightskylark) [inline js/data/odata/utils.js:366]: Nu Igoryok! (T213119)

**Review by pavelgruba — DISMISSED** [review]

File: js/core/utils/ajax.js [hunk 3]
```diff
@@ -163,7 +163,6 @@ var isCrossDomain = function(url) {
 
     try {
         urlAnchor.href = url;
-        urlAnchor.href = urlAnchor.href;
```
Comment (nightskylark) [inline js/core/utils/ajax.js]: I was interested in how such a code could be written. So I have made a small investigation.
https://github.com/DevExpress/DevExtreme/pull/1177/files#diff-5f87e0d504be9bdc9d62bcf19d73f6e8R150
Too large PR.

**Review by dxvladislavvolkov — DISMISSED** [review]

**Review by BingoRUS — DISMISSED** [review]
👌😉

File: js/core/utils/ajax.js (hunk 3 above)
Comment (babich-a) [reply js/core/utils/ajax.js]: Are you ready for magic?
```
var urlAnchor = document.createElement("a");
urlAnchor.href = "12";
// urlAnchor.href = urlAnchor.href;
console.log(urlAnchor.host);
```
Run this code with comment, and without it in my 'favorite' browser (IE11)

**Review by babich-a — CHANGES_REQUESTED** [review]

File: js/core/utils/ajax.js (hunk 3 above)
Comment (nightskylark) [reply js/core/utils/ajax.js]: OMG. It's the craziest thing I have ever seen. If so we need to left some note at least.

Comment (babich-a) [reply js/core/utils/ajax.js]: I agree. I have tested 3 browsers before my memory come back.

**Review by jsdmitry — DISMISSED** [review]

File: js/core/utils/ajax.js (hunk 3 above)
Comment (San4es) [reply js/core/utils/ajax.js]: Thank you @babich-a! I will fix this.

**Review by churkin — DISMISSED** [review]

**Review by babich-a — DISMISSED** [review]

**Review by jsdmitry — DISMISSED** [review]

**Review by nightskylark — APPROVED** [review]

**Review by babich-a — APPROVED** [review]

**Review by churkin — APPROVED** [review]

**Review by BingoRUS — APPROVED** [review]

**Review by dxvladislavvolkov — APPROVED** [review]

**Review by jsdmitry — APPROVED** [review]

**Review by pavelgruba — APPROVED** [review]

**Pull request merged**